pkg/provider/aws/action/eks/eks.go

```diff
@@ -0,0 +1,268 @@
+		return err
+	}
+	cs := manager.Stack{
+		StackName:           maptContext.StackNameByProject(stackName),
+		ProjectName:         maptContext.ProjectName(),
+		BackedURL:           maptContext.BackedURL(),
+		ProviderCredentials: aws.DefaultCredentials,
+		DeployFunc:          r.deployer,
+	}
+	sr, _ := manager.UpStack(cs)
+	return r.manageResults(sr)
+}
+
+func Destroy(ctx *maptContext.ContextArgs) error {
+	// Create mapt Context
+	logging.Debug("Destroy EKS")
+	if err := maptContext.Init(ctx, aws.Provider()); err != nil {
+		return err
+	}
+	return aws.DestroyStack(
+		aws.DestroyStackRequest{
+			BackedURL: maptContext.BackedURL(),
+			Stackname: stackName,
+	})
+}
+
+// Main function to deploy all requried resources to AWS
+func (r *EKSRequest) deployer(ctx *pulumi.Context) error {
+	// Read back the default VPC and public subnets, which we will use.
+	t := true
+	vpc, err := ec2.LookupVpc(ctx, &ec2.LookupVpcArgs{Default: &t})
+	if err != nil {
+		return err
+	}
+
+	subnet, err := ec2.GetSubnets(ctx, &ec2.GetSubnetsArgs{
+		Filters: []ec2.GetSubnetsFilter{
+			{Name: "vpc-id", Values: []string{vpc.Id}},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	eksRole, err := iam.NewRole(ctx, "eks-iam-eksRole", &iam.RoleArgs{
+		AssumeRolePolicy: pulumi.String(`{
+			"Version": "2008-10-17",
+			"Statement": [{
+					"Sid": "",
+					"Effect": "Allow",
+					"Principal": {
+							"Service": "eks.amazonaws.com"
+					},
+					"Action": "sts:AssumeRole"
+			}]
+	}`),
+	})
+	if err != nil {
+		return err
+	}
+	eksPolicies := []string{
+		"arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
+		"arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
+	}
+	for i, eksPolicy := range eksPolicies {
+		_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf("rpa-%d", i), &iam.RolePolicyAttachmentArgs{
+			PolicyArn: pulumi.String(eksPolicy),
+			Role:      eksRole.Name,
+		})
+		if err != nil {
+			return err
+		}
+	}
+	// Create the EC2 NodeGroup Role
+	nodeGroupRole, err := iam.NewRole(ctx, "nodegroup-iam-role", &iam.RoleArgs{
+		AssumeRolePolicy: pulumi.String(`{
+			"Version": "2012-10-17",
+			"Statement": [{
+					"Sid": "",
+					"Effect": "Allow",
+					"Principal": {
+							"Service": "ec2.amazonaws.com"
+					},
+					"Action": "sts:AssumeRole"
+			}]
+	}`),
+	})
+	if err != nil {
+		return err
+	}
+	nodeGroupPolicies := []string{
+		"arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+		"arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+		"arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+	}
+	for i, nodeGroupPolicy := range nodeGroupPolicies {
+		_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf("ngpa-%d", i), &iam.RolePolicyAttachmentArgs{
+			Role:      nodeGroupRole.Name,
+			PolicyArn: pulumi.String(nodeGroupPolicy),
+		})
+		if err != nil {
+			return err
+		}
+	}
+	// Create a Security Group that we can use to actually connect to our cluster
+	clusterSg, err := ec2.NewSecurityGroup(ctx, "cluster-sg", &ec2.SecurityGroupArgs{
```

Comment:
You can use the sg utility in the code to reduce this content:

https://github.com/redhat-developer/mapt/blob/main/pkg/provider/aws/action/fedora/fedora.go#L308

Reply:
Ok, I incorporated it, but I have issues with the networking and subnets. See the below comment